Sources/Core/Filters/Root/RootFilterViewController.swift

```diff
@@ -190,27 +180,22 @@ extension RootFilterViewController: VerticalListViewControllerDelegate {
     }
 }
 
-extension RootFilterViewController: SearchViewControllerDelegate {
-    func presentSearchViewController(_ searchViewController: SearchQueryViewController) {
-        add(searchViewController)
+extension RootFilterViewController: FreeTextFilterViewControllerDelegate {
+    func freeTextFilterViewControllerWillBeginEditing(_ viewController: FreeTextFilterViewController) {
+        add(viewController)
     }
 
-    func searchViewControllerDidCancelSearch(_ searchViewController: SearchQueryViewController) {
-        guard let searchFilter = searchFilter else {
-            return
-        }
-
-        selectionStore.removeValues(for: searchFilter)
+    func freeTextFilterViewControllerWillEndEditing(_ viewController: FreeTextFilterViewController) {
+        viewController.remove()
         tableView.reloadData()
     }
 
-    func searchViewController(_ searchViewController: SearchQueryViewController, didSelectQuery query: String) {
-        guard let searchFilter = searchFilter else {
-            return
+    func freeTextFilterViewController(_ viewController: FreeTextFilterViewController, didSelectValue value: String?, forFilter filter: Filter) {
+        if let value = value {
+            selectionStore.setValue(value, for: filter)
```

Comment:
You can just `selectionStore.setValue(value, for: filter)`, it will be removed if nil.

Reply:
Also, maybe `FreeTextFilterViewController` can be created with selection store and set it itself. That's what we do for range, map and list at least.

Reply:
Yes, we could do that